roles/wildfly_systemd/tasks/main.yml

```diff
@@ -15,16 +15,26 @@
   when:
     - wildfly_enable_yml_config
     - eap_enable is defined and eap_enable
+    - eap_apply_cp is defined and eap_apply_cp
 
 - name: Check arguments for yaml configuration
   ansible.builtin.assert:
     that:
-      - slurped_eap_version.content | b64decode is version('7.4.5', '>=')
+      - (slurped_eap_version.content | b64decode is version('7.4.5', '>='))
     quiet: true
     fail_msg: "Can only enable YAML configuration with EAP >= 7.4.5"
   when:
     - wildfly_enable_yml_config
     - eap_enable is defined and eap_enable
+    - wildfly_version.split('.')[:1] | join('.') is version('7', '==')
+
+- name: Check if YAML configuration extension is supported
+  ansible.builtin.assert:
```

Comment:
Can you split this tasks in two separated tasks (for readability?). One for wildfly (with version >= 26.0.1) and one for eap (with the other conditions and  eap_enabled is defined).